clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala

```diff
@@ -17,4 +49,712 @@ trait WgsCramTests { self: BaseIntegrationSpec =>
+        "--project",
+        project,
+        "--sample-alias",
+        sampleAlias,
+        "--include-deleted"
+      )
+      results <- Unmarshal(response).to[Seq[TransferWgsCramV1QueryOutput]]
+    } yield {
+      results.length should be(keysWithMetadata.length)
+      results.foldLeft(succeed) { (_, result) =>
+        result.project should be(project)
+        result.sampleAlias should be(sampleAlias)
+
+        val resultKey = TransferWgsCramV1Key(
+          location = result.location,
+          project = result.project,
+          sampleAlias = result.sampleAlias,
+          version = result.version
+        )
+
+        if (resultKey == deleteKey) {
+          result.documentStatus should be(Some(DocumentStatus.Deleted))
+        } else {
+          result.documentStatus should be(Some(DocumentStatus.Normal))
+        }
+      }
+    }
+  }
+
+  it should "move wgs-crams in GCP" in {
+    val project = s"project$randomId"
+    val sample = s"sample$randomId"
+    val version = 3
+
+    val fileContents = s"$randomId --- I am a dummy cram --- $randomId"
+    val cloudPath = rootTestStorageDir.resolve(
+      s"cram/$project/$sample/v$version/$randomId.cram"
+    )
+    val cloudPath2 = cloudPath.getParent.resolve(s"moved/$randomId.cram")
+
+    val key = TransferWgsCramV1Key(Location.GCP, project, sample, version)
+    val metadata =
+      TransferWgsCramV1Metadata(cramPath = Some(cloudPath.toUri.toString))
+
+    // Clio needs the metadata to be added before it can be moved.
+    val _ = Files.write(cloudPath, fileContents.getBytes)
+    val result = for {
+      _ <- runUpsertCram(key, metadata)
+      _ <- runClient(
+        ClioCommand.moveWgsCramName,
+        "--location",
+        Location.GCP.entryName,
+        "--project",
+        project,
+        "--sample-alias",
+        sample,
+        "--version",
+        version.toString,
+        "--destination",
+        cloudPath2.toUri.toString
+      )
+    } yield {
+      Files.exists(cloudPath) should be(false)
+      Files.exists(cloudPath2) should be(true)
+      new String(Files.readAllBytes(cloudPath2)) should be(fileContents)
+    }
+
+    result.andThen[Unit] {
+      case _ => {
+        // Without `val _ =`, the compiler complains about discarded non-Unit value.
+        val _ = Seq(cloudPath, cloudPath2).map(Files.deleteIfExists)
+      }
+    }
+  }
+
+  it should "move the cram, crai, md5, and some metrics files together in GCP" in {
+    val project = s"project$randomId"
+    val sample = s"sample$randomId"
+    val version = 3
+
+    val cramContents = s"$randomId --- I am a dummy cram --- $randomId"
+    val craiContents = s"$randomId --- I am a dummy crai --- $randomId"
+    val md5Contents = randomId
+    val alignmentMetricsContents =
+      s"$randomId --- I am dummy alignment metrics --- $randomId"
+    val fingerprintMetricsContents =
+      s"$randomId --- I am dummy fingerprinting metrics --- $randomId"
+
+    val cramName = s"$randomId.cram"
```

Comment:
There's gotta be some workaround for this kind of parallel construction in Scala? (No, I don't know how.)

Reply:
There is definitely a better way to do this.